src/WebJobs.Script/Description/Workers/WorkerFunctionDescriptorProvider.cs

```diff
@@ -54,6 +55,9 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
         {
             var parameters = await base.GetFunctionParametersAsync(functionInvoker, functionMetadata, triggerMetadata, methodAttributes, inputBindings, outputBindings);
 
+            // Add cancellation token
+            parameters.Add(new ParameterDescriptor(ScriptConstants.SystemCancellationTokenParameterName, typeof(CancellationToken)));
```

Comment:
Pardon my silly question :) Do we need to add it like this? If OOP function has CancellationToken param, will it be not added like other bindings/parameters automatically with existing code? Also where is this parameter (`_cancellationToken`) being used?

Reply:
I think @fabiocav might know the answer better, but from my understanding WebJobs has a cancellation token and this line is allowing us to grab the `_cancellationToken` param from WebJobs (this name is defined by WebJobs I think?). Once we have the cancellation token in the parameters object, it flows down to [InvokeCore](https://github.com/Azure/azure-functions-host/blob/422c5b4059edc005ecf94b7d4804155b35da5e61/src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs#L58) where we add the CT to the invocation context.